src/db/device/index.js

```diff
@@ -2,16 +2,29 @@ import deviceModel from './model';
 
 /**
  * @description Gets a single device from the database
- * @param {object} deviceDetails  The data to be used as filter
- * @param {string} deviceDetails._id
- * @param {string} deviceDetails.hospitalId
- * @param {string} deviceDetails.wardId
+ * @param {object} deviceMatch  The data to be used as filter
+ * @param {string} deviceMatch._id
+ * @param {string} deviceMatch.hospitalId
+ * @param {string} deviceMatch.wardId
  * @returns {Promise} A promise that resolves or reject to the result of the database operation
  */
-const getSingleDevice = async (deviceDetails) => {
-  const params = JSON.parse(JSON.stringify(deviceDetails));
+const getSingleDevice = async (deviceMatch) => {
   const Model = deviceModel;
-  return Model.findOne(params);
+  return Model.findOne(deviceMatch);
+};
+
+/**
+ * @description Gets all devices matching the query parameter
+ * @param {object} deviceMatch  The data to be used as filter for getting all devices
+ * @param {string} deviceMatch.hospitalId The hospitalId will used as the filter
+ * if the hospital_admin request for a device
```

Comment:
No need or extra details on who is making the request, the db module does not know about that